### PR TITLE
generation-encounters: Added encounter entity generation

### PR DESCRIPTION
**Pull request by wojtekbu**
* Refactored generation options calsses with lombok

### CODE REVIEW

File: src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java [hunk 1]
```diff
@@ -122,6 +130,9 @@ private void setOptions() {
                 case CommandConstants.VISITS_AMOUNT:
                     visitOption.setVisits(Long.parseLong(option.getValue()));
                     break;
+                case CommandConstants.ENCOUNTERS_AMOUNT:
+                    encounterOption.setEncounters(Long.parseLong(option.getValue()));
```
Comment (pawelcieszko) [inline src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java]: is there a possibility that option.getValue() == null?

File: src/main/java/org/openmrs/isanteplus/performancedata/model/Encounter.java [hunk 2]
```diff
@@ -49,23 +46,22 @@
...
     }
 
     @Builder
-    public Encounter(long id, long encounterType, long patientId, long encounterId,
+    public Encounter(long id, long encounterType, long patientId,
                      long locationId, long formId, LocalDateTime encounterDatetime,
                      long creator, LocalDateTime dateCreated, long voided, long voidedBy,
                      LocalDateTime dateVoided, String voidReason, long changedBy,
-                     LocalDateTime dateChanged, long visitId, UUID uuid) {
+                     LocalDateTime dateChanged, long visitId, String uuid) {
```
Comment (pawelcieszko) [inline src/main/java/org/openmrs/isanteplus/performancedata/model/Encounter.java]: I woudn't change UUID type to string if there is no need. Keeping UUID type helps in avoiding invalid type/value passing

File: src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java [hunk 3]
```diff
@@ -91,6 +95,10 @@ public long getVisits() {
         return visitOption.getVisits();
     }
 
+    public long getEncounters() {
```
Comment (pawelcieszko) [inline src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java]: shouldn't it be getEncountersNumber? or size?

File: src/main/java/org/openmrs/isanteplus/performancedata/options/model/EncounterOption.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+@AllArgsConstructor
+public class EncounterOption {
+    private static final long ENCOUNTERS = 1;
+
+    @Getter
+    @Setter
+    private long encounters;
```
Comment (pawelcieszko) [inline src/main/java/org/openmrs/isanteplus/performancedata/options/model/EncounterOption.java]: it is rather not a collection of encounters but their number - shouldn't we change it?

File: src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java (hunk 3 above)
Comment (wojtekbu) [reply src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java]: toString of UUID does not match mysql insert syntax. I can wrap it up in some custom class and override toString method, but String works just fine as well.

File: src/main/java/org/openmrs/isanteplus/performancedata/options/model/EncounterOption.java (hunk 4 above)
Comment (wojtekbu) [reply src/main/java/org/openmrs/isanteplus/performancedata/options/model/EncounterOption.java]: ok

File: src/main/java/org/openmrs/isanteplus/performancedata/model/Encounter.java (hunk 2 above)
Comment (wojtekbu) [reply src/main/java/org/openmrs/isanteplus/performancedata/model/Encounter.java]: No, the tool will stop an execution. I'll add a more precise message for the user.

File: src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java (hunk 1 above)
Comment (wojtekbu) [reply src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java]: No, the tool will stop an execution. I'll add a more precise message for the user.

File: src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java (hunk 3 above)
Comment (wojtekbu) [reply src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java]: ok

File: src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java [hunk 5]
```diff
@@ -96,7 +96,7 @@ public long getVisits() {
     }
 
     public long getEncounters() {
-        return encounterOption.getEncounters();
+        return encounterOption.getEncountersAmount();
```
Comment (pawelcieszko) [inline src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java]: number not amount

File: src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java [hunk 6]
```diff
@@ -131,7 +131,7 @@ private void setOptions() {
                     visitOption.setVisits(Long.parseLong(option.getValue()));
                     break;
                 case CommandConstants.ENCOUNTERS_AMOUNT:
-                    encounterOption.setEncounters(Long.parseLong(option.getValue()));
+                    encounterOption.setEncountersAmount(Long.parseLong(option.getValue()));
```
Comment (pawelcieszko) [inline src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java]: number not amount

File: src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java (hunk 3 above)
Comment (pawelcieszko) [reply src/main/java/org/openmrs/isanteplus/performancedata/options/GeneratorOptions.java]: up?

File: src/main/java/org/openmrs/isanteplus/performancedata/model/Encounter.java (hunk 2 above)
Comment (pawelcieszko) [reply src/main/java/org/openmrs/isanteplus/performancedata/model/Encounter.java]: ok

**Pull request merged**